presto-main/src/main/java/com/facebook/presto/type/RealOperators.java

```diff
@@ -165,7 +165,7 @@ public static boolean between(@SqlType(StandardTypes.REAL) long value, @SqlType(
     @SqlType(StandardTypes.BIGINT)
     public static long hashCode(@SqlType(StandardTypes.REAL) long value)
     {
-        return AbstractIntType.hash((int) value);
+        return AbstractIntType.hash(floatToIntBits(intBitsToFloat((int) value)));
```

Comment:
could you please add a test that captures the difference?

Reply:
Will submit another PR to add the unit test. Thanks for pointing out!

Reply:
Thanks!